File: 724-find-pivot-index/724-find-pivot-index.c


int pivotIndex(int* nums, int numsSize){
    int sum = 0, leftsum = 0;
    for(int i = 0; i < numsSize; i++){sum+=nums[i];}
    for(int i = 0; i < numsSize; i++){
        if(sum - leftsum - nums[i] == leftsum){return i;}
        leftsum += nums[i];
    }
    return -1;
}

File: 724-find-pivot-index/724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int l = nums.length, sum = 0, lsum = 0;
        for(int i = 0; i < l; i++){sum+=nums[i];}
        for(int i = 0; i < l; i++){
            if(sum - lsum - nums[i] == lsum)
                return i;
            lsum+=nums[i];
        }
        return -1;
    }
}